pythran/pythonic/types/dict.hpp

```diff
@@ -609,93 +609,58 @@ namespace std
     return d[I];
   }
 }
-
 #ifdef ENABLE_PYTHON_MODULE
 
-#include "pythonic/python/register_once.hpp"
-#include "pythonic/python/extract.hpp"
-#include <boost/python/dict.hpp>
-#include <boost/python/object.hpp>
+#include "pythonic/python/core.hpp"
 
 namespace pythonic
 {
 
   template <typename K, typename V>
-  python_to_pythran<types::dict<K, V>>::python_to_pythran()
-  {
-    python_to_pythran<K>();
-    python_to_pythran<V>();
-    static bool registered = false;
-    if (not registered) {
-      registered = true;
-      boost::python::converter::registry::push_back(
-          &convertible, &construct,
-          boost::python::type_id<types::dict<K, V>>());
+  struct to_python<types::dict<K, V>> {
+    static PyObject *convert(types::dict<K, V> const &v)
+    {
+      PyObject *ret = PyDict_New();
+      for (auto kv = v.item_begin(); kv != v.item_end(); ++kv)
+        PyDict_SetItem(ret, ::to_python(kv->first), ::to_python(kv->second));
```

Comment:
So, omp can improve the to_python conversion? Or there is the GIL everywhere (so it will be slow anyway...)